libtextsecure/message_receiver.js

```diff
@@ -1048,12 +1066,10 @@ MessageReceiver.prototype.extend({
         // ref: libsignal-protocol.js:36120
         envelope.type === textsecure.protobuf.Envelope.Type.PREKEY_BUNDLE
       ) {
-      // We know for sure that keys are exchanged
-      if (conversation) {
+      // If we get a cipher and we're already friends
+      // then we set our key exchange to complete
+      if (conversation && conversation.isFriend()) {
```

Comment:
This will fail when we get a friend request accept as at that time `isFriend` would be false.